src/main/java/fr/insee/pogues/webservice/rest/PoguesTransforms.java

```diff
@@ -506,11 +512,17 @@ private ResponseEntity<StreamingResponseBody> transform(InputStream request, Tra
 			try {
 				transformer.transform(request, output, null, questionnaire);
 			} catch (Exception e) {
-				e.printStackTrace();
-				throw new PoguesException(500, "Transformation error", e.getMessage());
+				logger.error(e.getCause().getMessage());
+				throw new PoguesException(500, e.getCause().getClass().getSimpleName(), e.getCause().getMessage());
 			}
 		};
 		return ResponseEntity.status(HttpStatus.OK).contentType(type).body(stream);
 	}
 
+	@ExceptionHandler(PoguesException.class)
+	public ResponseEntity<ApiError> handlePoguesException(PoguesException pe) {
+		ApiError apiErrorResponse = new ApiError(pe.getStatus(), pe.getMessage(), pe.getDetails());
+		return new ResponseEntity<>(apiErrorResponse, HttpStatus.valueOf(pe.getStatus()));
+	}
+

```

Comment:
I think it would be good to group exception handlers somewhere, this one feels a bit hidden here